The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11161527492/job/31024283293) [(plain)](https://github.com/rust-lang/rust/commit/5d7f66dbf3364deabec252f74cd54b97a6582a31/checks/31024283293/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
------
 > importing cache manifest from ghcr.io/rust-lang/rust-ci-cache:20950f2ccee3dff53a038adf5c1cf05231c0b30772617126a5f6478a66316a29cba9aead69f7bb0004886d32c1f8e6287542cf6d25130711c82d16a66201d4fe:
------
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
................

failures:

---- [codegen] tests/codegen/binary-heap-peek-mut-pop-no-panic.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-18/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/binary-heap-peek-mut-pop-no-panic/binary-heap-peek-mut-pop-no-panic.ll" "/checkout/tests/codegen/binary-heap-peek-mut-pop-no-panic.rs" "--check-prefix=CHECK" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
--- stderr -------------------------------
/checkout/tests/codegen/binary-heap-peek-mut-pop-no-panic.rs:9:16: error: CHECK-NOT: excluded string found in input
 // CHECK-NOT: call {{.*}}drop_in_place
               ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/binary-heap-peek-mut-pop-no-panic/binary-heap-peek-mut-pop-no-panic.ll:420:3: note: found here
; call core::ptr::drop_in_place<alloc::collections::binary_heap::PeekMut<u32>>


Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/binary-heap-peek-mut-pop-no-panic/binary-heap-peek-mut-pop-no-panic.ll
Check file: /checkout/tests/codegen/binary-heap-peek-mut-pop-no-panic.rs

-dump-input=help explains the following input dump.
Input was:
<<<<<<
       .
       .
       .
       .
     320:  store i32 %_4.i.i.i.i.i.i, ptr %hole_ptr.i.i.i.i.i.i, align 4, !noalias !62 
     321:  %_7.not.i.i.i.i.i = icmp ult i64 %_14.0.i.i.i.i.i, 2 
     322:  br i1 %_7.not.i.i.i.i.i, label %"_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$19sift_down_to_bottom17hdc4c8d917c6133b1E.exit.i.i.i", label %bb2.i26.i.i.i.i 
     323:  
     324: cleanup.body.thread.i.i.i.i: ; preds = %bb4.i.i.i.i.i.i.invoke 
     325:  %16 = landingpad { ptr, i32 } 
     326:  cleanup 
     327:  br label %bb20.i.i.i.i 
     328:  
     329: bb6.i.i.i.i: ; preds = %bb1.i.i.i.i.i 
     330:  %_11.i.i.i.i.i = getelementptr inbounds i32, ptr %self1.i.i.i, i64 %child.sroa.0.0101.i.i.i.i 
     331:  %_30.0.i.i.i.i = add i64 %child.sroa.0.0.in100.i.i.i.i, 2 
     332:  %_3.not.i29.i.i.i.i = icmp eq i64 %hole.sroa.16.099.i.i.i.i, %_30.0.i.i.i.i 
     333:  br i1 %_3.not.i29.i.i.i.i, label %bb4.i45.i.i.i.i.invoke, label %bb1.i30.i.i.i.i 
     334:  
     335: bb1.i30.i.i.i.i: ; preds = %bb6.i.i.i.i 
     336:  %_6.i32.i.i.i.i = icmp ugt i64 %_3.0.i.i.i, %_30.0.i.i.i.i 
     337:  br i1 %_6.i32.i.i.i.i, label %bb27.i.i.i.i, label %bb4.i45.i.i.i.i.invoke 
     338:  
     339: bb27.i.i.i.i: ; preds = %bb1.i30.i.i.i.i 
     340:  %_11.i35.i.i.i.i = getelementptr inbounds i32, ptr %self1.i.i.i, i64 %_30.0.i.i.i.i 
     341:  tail call void @llvm.experimental.noalias.scope.decl(metadata !65) 
     342:  tail call void @llvm.experimental.noalias.scope.decl(metadata !68) 
     343:  %_3.i.i.i.i.i = load i32, ptr %_11.i.i.i.i.i, align 4, !alias.scope !65, !noalias !70, !noundef !6 
     344:  %_4.i.i.i.i.i = load i32, ptr %_11.i35.i.i.i.i, align 4, !alias.scope !68, !noalias !71, !noundef !6 
     345:  %_0.i.i.i.i.i = icmp ule i32 %_3.i.i.i.i.i, %_4.i.i.i.i.i 
     346:  %_18.i.i.i.i = zext i1 %_0.i.i.i.i.i to i64 
     347:  %_31.0.i.i.i.i = add nuw i64 %child.sroa.0.0101.i.i.i.i, %_18.i.i.i.i 
     348:  %_3.not.i41.i.i.i.i = icmp eq i64 %hole.sroa.16.099.i.i.i.i, %_31.0.i.i.i.i 
     349:  br i1 %_3.not.i41.i.i.i.i, label %bb4.i45.i.i.i.i.invoke, label %bb1.i42.i.i.i.i 
     350:  
     351: bb1.i42.i.i.i.i: ; preds = %bb27.i.i.i.i 
     352:  %_6.i44.i.i.i.i = icmp ugt i64 %_3.0.i.i.i, %_31.0.i.i.i.i 
     353:  br i1 %_6.i44.i.i.i.i, label %bb10.i.i.i.i, label %bb4.i45.i.i.i.i.invoke 
     354:  
     355: bb4.i45.i.i.i.i.invoke: ; preds = %bb1.i42.i.i.i.i, %bb27.i.i.i.i, %bb1.i30.i.i.i.i, %bb6.i.i.i.i, %bb1.i.i.i.i.i 
     356:  %17 = phi ptr [ @alloc_a0c46c1cfb162d5e836300b91462d82f, %bb1.i.i.i.i.i ], [ @alloc_f4f54628c8d1c2ce33ac207dfec4fe85, %bb6.i.i.i.i ], [ @alloc_a0c46c1cfb162d5e836300b91462d82f, %bb1.i30.i.i.i.i ], [ @alloc_f4f54628c8d1c2ce33ac207dfec4fe85, %bb27.i.i.i.i ], [ @alloc_a0c46c1cfb162d5e836300b91462d82f, %bb1.i42.i.i.i.i ] 
     357:  %18 = phi i64 [ 41, %bb1.i.i.i.i.i ], [ 35, %bb6.i.i.i.i ], [ 41, %bb1.i30.i.i.i.i ], [ 35, %bb27.i.i.i.i ], [ 41, %bb1.i42.i.i.i.i ] 
     358:  %19 = phi ptr [ @alloc_bda55c97e1a636a0b186dd1984e1bc0b, %bb1.i.i.i.i.i ], [ @alloc_b8d31b75f06593412cf852bcac4c5d7c, %bb6.i.i.i.i ], [ @alloc_bda55c97e1a636a0b186dd1984e1bc0b, %bb1.i30.i.i.i.i ], [ @alloc_10a0ecbdce71f72fa90593e92904e1e6, %bb27.i.i.i.i ], [ @alloc_e0cfdb897879ac7908de6fbeda10ffc1, %bb1.i42.i.i.i.i ] 
     359: ; invoke core::panicking::panic 
     360:  invoke void @_RNvNtCsmhgpoCUEUe_4core9panicking5panic(ptr noalias noundef nonnull readonly align 1 %17, i64 noundef %18, ptr noalias noundef nonnull readonly align 8 dereferenceable(24) %19) #8 
     361:  to label %bb4.i45.i.i.i.i.cont unwind label %bb21.i.i.i.i, !noalias !53 
     362:  
     363: bb4.i45.i.i.i.i.cont: ; preds = %bb4.i45.i.i.i.i.invoke 
     364:  unreachable 
     365:  
     366: bb10.i.i.i.i: ; preds = %bb1.i42.i.i.i.i 
     367:  %_11.i47.i.i.i.i = getelementptr inbounds i32, ptr %self1.i.i.i, i64 %_31.0.i.i.i.i 
     368:  %hole_ptr.i48.i.i.i.i = getelementptr inbounds i32, ptr %self1.i.i.i, i64 %hole.sroa.16.099.i.i.i.i 
     369:  %20 = load i32, ptr %_11.i47.i.i.i.i, align 4, !noalias !72 
     370:  store i32 %20, ptr %hole_ptr.i48.i.i.i.i, align 4, !noalias !72 
     371:  %child.sroa.0.0.in.i.i.i.i = shl i64 %_31.0.i.i.i.i, 1 
     372:  %child.sroa.0.0.i.i.i.i = or disjoint i64 %child.sroa.0.0.in.i.i.i.i, 1 
     373:  %_15.not.not.i.i.i.i = icmp ult i64 %child.sroa.0.0.in.i.i.i.i, %10 
     374:  br i1 %_15.not.not.i.i.i.i, label %bb5.i.i.i.i, label %bb13.i.i.i.i 
     375:  
     376: bb20.i.i.i.i: ; preds = %bb21.i.i.i.i, %cleanup.body.thread.i.i.i.i 
     377:  %hole.sroa.14.033.i142.sink.i.i.i.i = phi i64 [ %hole.sroa.14.033.i.i.i.i.i, %cleanup.body.thread.i.i.i.i ], [ %hole.sroa.16.097.i.i.i.i, %bb21.i.i.i.i ] 
     378:  %eh.lpad-body66.i.i.i.i = phi { ptr, i32 } [ %16, %cleanup.body.thread.i.i.i.i ], [ %lpad.thr_comm.i.i.i.i, %bb21.i.i.i.i ] 
     379:  %21 = getelementptr inbounds i32, ptr %self1.i.i.i, i64 %hole.sroa.14.033.i142.sink.i.i.i.i 
     380:  store i32 %_7.i.i.i, ptr %21, align 4, !noalias !53 
     381:  br label %cleanup.body.i 
     382:  
     383: bb21.i.i.i.i: ; preds = %bb4.i45.i.i.i.i.invoke, %bb2.i23.invoke.i.i.i.i, %bb2.i14.i.i.i.i 
     384:  %hole.sroa.16.097.i.i.i.i = phi i64 [ %child.sroa.0.0101.i.i.i.i, %bb2.i14.i.i.i.i ], [ %_31.0.i.i.i.i, %bb2.i23.invoke.i.i.i.i ], [ %hole.sroa.16.099.i.i.i.i, %bb4.i45.i.i.i.i.invoke ] 
     385:  %lpad.thr_comm.i.i.i.i = landingpad { ptr, i32 } 
     386:  cleanup 
     387:  br label %bb20.i.i.i.i 
     388:  
     389: "_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$19sift_down_to_bottom17hdc4c8d917c6133b1E.exit.i.i.i": ; preds = %"_ZN5alloc11collections11binary_heap13Hole$LT$T$GT$7move_to17ha81133b8e5d472b2E.exit.i.i.i.i.i", %bb16.i.i.i.i.i, %bb13.thread.i.i.i.i 
     390:  %hole.sroa.14.0.lcssa.i.i.i.i.i = phi i64 [ 0, %bb13.thread.i.i.i.i ], [ %parent3.i.i.i.i.i, %"_ZN5alloc11collections11binary_heap13Hole$LT$T$GT$7move_to17ha81133b8e5d472b2E.exit.i.i.i.i.i" ], [ %hole.sroa.14.033.i.i.i.i.i, %bb16.i.i.i.i.i ] 
     391:  %22 = getelementptr inbounds i32, ptr %self1.i.i.i, i64 %hole.sroa.14.0.lcssa.i.i.i.i.i 
     392:  store i32 %_7.i.i.i, ptr %22, align 4, !noalias !75 
     393:  br label %"_ZN5alloc11collections11binary_heap20PeekMut$LT$T$C$A$GT$3pop17h6021eb16eca18132E.exit" 
     394:  
     395: bb9.i: ; preds = %bb1.i 
     396: ; invoke core::panicking::panic 
     397:  invoke void @_RNvNtCsmhgpoCUEUe_4core9panicking5panic(ptr noalias noundef nonnull readonly align 1 @alloc_00c4bc2c43db60dab26d25d7c2e3bd2a, i64 noundef 44, ptr noalias noundef nonnull readonly align 8 dereferenceable(24) @alloc_6b4df25c6de2658f0133179eeb3ac2bd) #8 
     398:  to label %unreachable.i unwind label %cleanup.i 
     399:  
     400: bb2.thread.i: ; preds = %bb1.i 
     401:  %23 = getelementptr inbounds i8, ptr %peek_mut.1, i64 16 
     402:  store i64 %peek_mut.0, ptr %23, align 8, !alias.scope !35 
     403:  br label %bb5.i.i 
     404:  
     405: unreachable.i: ; preds = %bb9.i 
     406:  unreachable 
     407:  
     408: terminate.i: ; preds = %cleanup.body.i 
     409:  %24 = landingpad { ptr, i32 } 
     410:  filter [0 x ptr] zeroinitializer 
     411: ; call core::panicking::panic_in_cleanup 
     412:  tail call void @_RNvNtCsmhgpoCUEUe_4core9panicking16panic_in_cleanup() #10 
     413:  unreachable 
     414:  
     415: bb6.i: ; preds = %cleanup.body.i 
     416:  resume { ptr, i32 } %eh.lpad-body.i 
     417:  
     418: "_ZN5alloc11collections11binary_heap20PeekMut$LT$T$C$A$GT$3pop17h6021eb16eca18132E.exit": ; preds = %bb2.i, %bb5.i.i, %"_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$19sift_down_to_bottom17hdc4c8d917c6133b1E.exit.i.i.i" 
     419:  %_0.sroa.3.0.i.i = phi i32 [ %_7.i.i.i, %bb5.i.i ], [ %item.sroa.0.0.copyload.i.i.i, %"_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$19sift_down_to_bottom17hdc4c8d917c6133b1E.exit.i.i.i" ], [ undef, %bb2.i ] 
     420: ; call core::ptr::drop_in_place<alloc::collections::binary_heap::PeekMut<u32>> 
not:9       !~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                 error: no match expected
     421:  call fastcc void @"_ZN4core3ptr72drop_in_place$LT$alloc..collections..binary_heap..PeekMut$LT$u32$GT$$GT$17h26af164bd285adbcE"(ptr noalias noundef nonnull align 8 dereferenceable(16) %this.i) 
     422:  call void @llvm.lifetime.end.p0(i64 16, ptr nonnull %this.i) 
     423:  ret i32 %_0.sroa.3.0.i.i 
     424: } 
     425:  
     426: ; Function Attrs: nounwind nonlazybind uwtable 
     427: declare noundef i32 @rust_eh_personality(i32 noundef, i32 noundef, i64 noundef, ptr noundef, ptr noundef) unnamed_addr #1 
     429: ; core::panicking::panic 
     429: ; core::panicking::panic 
     430: ; Function Attrs: cold noinline noreturn nonlazybind uwtable 
     431: declare void @_RNvNtCsmhgpoCUEUe_4core9panicking5panic(ptr noalias noundef nonnull readonly align 1, i64 noundef, ptr noalias noundef readonly align 8 dereferenceable(24)) unnamed_addr #2 
     432:  
     433: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) 
     434: declare void @llvm.assume(i1 noundef) #3 
     436: ; core::panicking::panic_in_cleanup 
     436: ; core::panicking::panic_in_cleanup 
     437: ; Function Attrs: cold minsize noinline noreturn nounwind nonlazybind optsize uwtable 
     438: declare void @_RNvNtCsmhgpoCUEUe_4core9panicking16panic_in_cleanup() unnamed_addr #4 
     439:  
     440: ; Function Attrs: mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none) 
     441: declare i64 @llvm.usub.sat.i64(i64, i64) #5 
     442:  
     443: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
     444: declare void @llvm.lifetime.start.p0(i64 immarg, ptr nocapture) #6 
     445:  
     446: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
     447: declare void @llvm.lifetime.end.p0(i64 immarg, ptr nocapture) #6 
     448:  
     449: ; Function Attrs: nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: readwrite) 
     450: declare void @llvm.experimental.noalias.scope.decl(metadata) #7 
     451:  
     452: attributes #0 = { nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
     453: attributes #1 = { nounwind nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
     454: attributes #2 = { cold noinline noreturn nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
     455: attributes #3 = { mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) } 
     456: attributes #4 = { cold minsize noinline noreturn nounwind nonlazybind optsize uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
     457: attributes #5 = { mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none) } 
     458: attributes #6 = { mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) } 
     459: attributes #7 = { nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: readwrite) } 
     460: attributes #8 = { noreturn } 
     461: attributes #9 = { cold } 
     462: attributes #10 = { cold noreturn nounwind } 
     463:  
     464: !llvm.module.flags = !{!0, !1} 
     465: !llvm.ident = !{!2} 
     466:  
     467: !0 = !{i32 8, !"PIC Level", i32 2} 
     468: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
     469: !2 = !{!"rustc version 1.83.0-nightly (6aaf8d542 2024-10-03)"} 
     470: !3 = !{!4} 
     471: !4 = distinct !{!4, !5, !"_ZN95_$LT$alloc..collections..binary_heap..PeekMut$LT$T$C$A$GT$$u20$as$u20$core..ops..drop..Drop$GT$4drop17hdfce3a6bfd0c8f65E: %self"} 
     472: !5 = distinct !{!5, !"_ZN95_$LT$alloc..collections..binary_heap..PeekMut$LT$T$C$A$GT$$u20$as$u20$core..ops..drop..Drop$GT$4drop17hdfce3a6bfd0c8f65E"} 
     473: !6 = !{} 
     474: !7 = !{i64 8} 
     475: !8 = !{!9} 
     476: !9 = distinct !{!9, !10, !"_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$15sift_down_range17h959aed161181913aE: %self"} 
     477: !10 = distinct !{!10, !"_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$15sift_down_range17h959aed161181913aE"} 
     478: !11 = !{!12} 
     479: !12 = distinct !{!12, !13, !"_ZN5alloc11collections11binary_heap13Hole$LT$T$GT$3new17he701e954eb1bc10cE: %data.0"} 
     480: !13 = distinct !{!13, !"_ZN5alloc11collections11binary_heap13Hole$LT$T$GT$3new17he701e954eb1bc10cE"} 
     481: !14 = !{!15, !9, !4} 
     482: !15 = distinct !{!15, !13, !"_ZN5alloc11collections11binary_heap13Hole$LT$T$GT$3new17he701e954eb1bc10cE: %_0"} 
     483: !16 = !{!9, !4} 
     484: !17 = !{!18, !9, !4} 
     485: !18 = distinct !{!18, !19, !"_ZN5alloc11collections11binary_heap13Hole$LT$T$GT$7move_to17ha81133b8e5d472b2E: %self"} 
     486: !19 = distinct !{!19, !"_ZN5alloc11collections11binary_heap13Hole$LT$T$GT$7move_to17ha81133b8e5d472b2E"} 
     487: !20 = !{!21} 
     488: !21 = distinct !{!21, !22, !"_RNvXs10_NtNtCsmhgpoCUEUe_4core3cmp5implsmNtB8_10PartialOrd2le: %self"} 
     489: !22 = distinct !{!22, !"_RNvXs10_NtNtCsmhgpoCUEUe_4core3cmp5implsmNtB8_10PartialOrd2le"} 
     490: !23 = !{!24} 
     491: !24 = distinct !{!24, !22, !"_RNvXs10_NtNtCsmhgpoCUEUe_4core3cmp5implsmNtB8_10PartialOrd2le: %other"} 
     492: !25 = !{!24, !9, !4} 
     493: !26 = !{!21, !9, !4} 
     494: !27 = !{!28, !9, !4} 
     495: !28 = distinct !{!28, !29, !"_ZN5alloc11collections11binary_heap13Hole$LT$T$GT$7move_to17ha81133b8e5d472b2E: %self"} 
     496: !29 = distinct !{!29, !"_ZN5alloc11collections11binary_heap13Hole$LT$T$GT$7move_to17ha81133b8e5d472b2E"} 
     497: !30 = !{!31, !33, !9, !4} 
     498: !31 = distinct !{!31, !32, !"_ZN88_$LT$alloc..collections..binary_heap..Hole$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$4drop17hf090e17adc905aa7E: %self"} 
     499: !32 = distinct !{!32, !"_ZN88_$LT$alloc..collections..binary_heap..Hole$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$4drop17hf090e17adc905aa7E"} 
     500: !33 = distinct !{!33, !34, !"_ZN4core3ptr69drop_in_place$LT$alloc..collections..binary_heap..Hole$LT$u32$GT$$GT$17h88461bbb53549c73E: %_1"} 
     501: !34 = distinct !{!34, !"_ZN4core3ptr69drop_in_place$LT$alloc..collections..binary_heap..Hole$LT$u32$GT$$GT$17h88461bbb53549c73E"} 
     502: !35 = !{!36} 
     503: !36 = distinct !{!36, !37, !"_ZN5alloc11collections11binary_heap20PeekMut$LT$T$C$A$GT$3pop17h6021eb16eca18132E: argument 0"} 
     504: !37 = distinct !{!37, !"_ZN5alloc11collections11binary_heap20PeekMut$LT$T$C$A$GT$3pop17h6021eb16eca18132E"} 
     505: !38 = !{!39} 
     506: !39 = distinct !{!39, !40, !"_ZN4core3mem7replace17hc15798325acfafbdE: %dest"} 
     507: !40 = distinct !{!40, !"_ZN4core3mem7replace17hc15798325acfafbdE"} 
     508: !41 = !{!42, !44, !36} 
     509: !42 = distinct !{!42, !43, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$3pop17hb7c0336941d8ce79E: %self"} 
     510: !43 = distinct !{!43, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$3pop17hb7c0336941d8ce79E"} 
     511: !44 = distinct !{!44, !45, !"_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$3pop17h298d855d59d4436dE: %self"} 
     512: !45 = distinct !{!45, !"_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$3pop17h298d855d59d4436dE"} 
     513: !46 = !{!44} 
     514: !47 = !{!42} 
     515: !48 = !{!44, !36} 
     516: !49 = !{!42, !44} 
     517: !50 = !{!51, !44} 
     518: !51 = distinct !{!51, !52, !"_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$3pop28_$u7b$$u7b$closure$u7d$$u7d$17h8b3b83a4a39a8283E: %_1"} 
     519: !52 = distinct !{!52, !"_ZN5alloc11collections11binary_heap23BinaryHeap$LT$T$C$A$GT$3pop28_$u7b$$u7b$closure$u7d$$u7d$17h8b3b83a4a39a8283E"} 
     520: !53 = !{!54, !51, !44} 
       .
       .
>>>>>>
------------------------------------------
```

</details>

--- a/library/alloc/src/collections/binary_heap/mod.rs
+++ b/library/alloc/src/collections/binary_heap/mod.rs
@@ -374,7 +374,10 @@ impl<'a, T: Ord, A: Allocator> PeekMut<'a, T, A> {
             // the caller could've mutated the element. It is removed from the
             // heap on the next line and pop() is not sensitive to its value.
         }
-        this.heap.pop().unwrap()
+
+        // SAFETY: Have a `PeekMut` element proves that the associated binary heap being non-empty,
+        // so the `pop` operation will not fail.
+        unsafe { this.heap.pop().unwrap_unchecked() }
     }
 }
 

--- a/tests/codegen/binary-heap-peek-mut-pop-no-panic.rs
+++ b/tests/codegen/binary-heap-peek-mut-pop-no-panic.rs
@@ -1,0 +1,13 @@
+//@ compile-flags: -O
+//@ ignore-debug
+#![crate_type = "lib"]
+
+use std::collections::binary_heap::PeekMut;
+
+// CHECK-LABEL: @peek_mut_pop
+#[no_mangle]
+pub fn peek_mut_pop(peek_mut: PeekMut<u32>) -> u32 {
+    // CHECK-NOT: panic
+    // CHECK-NOT: unwrap_failed
+    PeekMut::pop(peek_mut)
+}

